### PR TITLE
Feature double banana

### PR DESCRIPTION
**Pull request by Maaffaa**
Intersection module added to Graph class.
t-sum method added to Graph class.
Add generalized double banana function in the graphBD.  It uses the t-sum method.

### CODE REVIEW

File: pyrigi/graph.py [hunk 1]
```diff
@@ -2112,6 +2112,62 @@ def _resolve_edge_colors(
...
+        >>> G = Graph([[0,1],[1,2],[2,3],[3,1]])
+        >>> H.sum_t(G, [1, 2], 3)
+        Graph with vertices [0, 1, 2, 3, 4] and edges [[0, 1], [1, 3], [2, 3], [3, 4]]
+        """
+        if edge not in self.edges or edge not in G2.edges:
+            raise ValueError(
+                f"The edge {edge} is not in the intersection of the graphs."
+            )
+        # check if the intersection is a t-complete graph
+        if self.intersection(G2).is_isomorphic(Graph(nx.complete_graph(t))) == False:
```
Comment (MatteoGallet) [inline pyrigi/graph.py:2142]: Here I would suggest to use

```
        if not self.intersection(G2).is_isomorphic(Graph(nx.complete_graph(t))):
```

**Review by MatteoGallet — COMMENTED** [review]
Please:
* merge the `dev` branch into this one and solve the conflicts;
* add the rest of the mathematical theory about generalized double bananas described by Anthony Nixon on the Zulip channel.

File: pyrigi/graph.py [hunk 2]
```diff
@@ -2112,6 +2112,62 @@ def _resolve_edge_colors(
...
+        ----------
+        G2: Graph
+
+        Examples
+        --------
+        >>> H = Graph([[1,2],[2,3],[3,1],[3,4]])
+        >>> G = Graph([[0,1],[1,2],[2,3],[3,1]])
+        >>> G.intersection(H)
+        Graph with vertices [1, 2, 3] and edges [[1, 2], [1, 3], [2, 3]]
+        """
```
Comment (MatteoGallet) [inline pyrigi/graph.py:2165]: Please add an example where the two graphs share also isolated vertices.

File: pyrigi/graph.py (hunk 1 above)
Comment (Maaffaa) [reply pyrigi/graph.py:2142]: I just changed this.

**Comment by Maaffaa** [comment]
> * add the rest of the mathematical theory about generalized double bananas described by Anthony Nixon on the Zulip channel.

Now I am doing this part, then I would do the merge.

**Review by MatteoGallet — APPROVED** [review]

**Pull request merged**